modules/quality_control/php/quality_control_behavioral.class.inc

```diff
@@ -0,0 +1,187 @@
+ *
+ * @category Module
+ * @package  Quality_Control
+ * @author   Liza Levitis <llevitis.mcin@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/CCNA/
+ */
+class Quality_Control_Behavioral extends \NDB_Menu_Filter
+{
+    public $hasWritePermission = false;
+    public $AjaxModule         = true;
+    public $fieldOptions;
+
+    /**
+     * Check user permissions
+     *
+     * @param \User $user UserID
+     *
+     * @return bool
+     * @throws \ConfigurationException
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        //create user object
+        return $user->hasPermission('data_team_helper');
+    }
+
+    /**
+     * Create a form to filter behavioral QC by various criteria
+     *
+     * @return void
+     * @throws \ConfigurationException
+     * @throws \DatabaseException
+     * @throws \Exception
+     */
+    function _setFilterForm(): void
+    {
+        $user      = \User::singleton();
+        $db        = \Database::singleton();
+        $siteList  = array();
+        $visitList = \Utility::getVisitList();
+        // allow to view all sites data through filter
+        if ($user->hasPermission('access_all_profiles')) {
+            $siteList = \Utility::getSiteList(false);
+            // Index sites using their names (used to filter react tables)
+            foreach ($siteList as $key => $site) {
+                unset($siteList[$key]);
+                $siteList[$site] = $site;
+            }
+        } else {
+            // allow only to view own site data
+            $siteIDs = $user->getData('CenterIDs');
+            foreach ($siteIDs as $val) {
+                $site = &Site::singleton($val);
+                if ($site->isStudySite()) {
+                    $siteList[$site->getCenterName()] = $site->getCenterName();
+                }
+            }
+        }
+        $instrumentList   = [];
+        $instrumentsQuery = $db->pselect(
+            "SELECT Test_name, Full_name FROM test_names ORDER BY Test_name",
+            []
+        );
+        foreach ($instrumentsQuery as $instrument) {
+            $instrumentList[$instrument['Full_name']] = $instrument['Full_name'];
+        }
+        $projectList = \Utility::getProjectList();
+        // Form Elements
+        $this->addSelect('visitLabel', 'Visit Label', $visitList);
+        $this->addSelect('instrument', 'Instruments', $instrumentList);
+        $this->addBasicText('candId', 'DCCID', ["size" => 9, "maxlength" => 7]);
+        $this->addBasicText('pSCID', 'PSCID', ["size" => 9, "maxlength" => 7]);
+        $this->addSelect('site', 'Site', $siteList);
+        $this->addSelect('project', 'Project', $projectList);
+
+        $hiddenOptions = [];
+        if ($this->hasHidePermission) {
+            $hiddenOptions = array(
+                              0 => 'Visible only',
+                              1 => 'Hidden only',
+                             );
+        }
+
```

Comment:
Lines `88`-`102` are no longer necessary because they are being handled by the front end